doc/languages-frameworks/nim.section.md

```diff
@@ -1,82 +1,73 @@
 # Nim {#nim}
 
-## Overview {#nim-overview}
-
-The Nim compiler, a builder function, and some packaged libraries are available
-in Nixpkgs. Until now each compiler release has been effectively backwards
-compatible so only the latest version is available.
-
-## Nim program packages in Nixpkgs {#nim-program-packages-in-nixpkgs}
-
-Nim programs can be built using `nimPackages.buildNimPackage`. In the
-case of packages not containing exported library code the attribute
-`nimBinOnly` should be set to `true`.
+The Nim compiler and a builder function is available.
+Nim programs are built using `buildNimPackage` and a lockfile containing Nim dependencies.
 
 The following example shows a Nim program that depends only on Nim libraries:
-
 ```nix
-{ lib, nimPackages, fetchFromGitHub }:
+{ lib, buildNimPackage, fetchFromGitHub }:
 
-nimPackages.buildNimPackage (finalAttrs: {
+buildNimPackage (finalAttrs: {
   pname = "ttop";
-  version = "1.0.1";
-  nimBinOnly = true;
+  version = "1.2.7";
 
   src = fetchFromGitHub {
     owner = "inv2004";
     repo = "ttop";
     rev = "v${finalAttrs.version}";
-    hash = "sha256-x4Uczksh6p3XX/IMrOFtBxIleVHdAPX9e8n32VAUTC4=";
+    hash = "sha256-oPdaUqh6eN1X5kAYVvevOndkB/xnQng9QVLX9bu5P5E=";
   };
 
-  buildInputs = with nimPackages; [ asciigraph illwill parsetoml zippy ];
+  lockFile = ./lock.json;
 
+  nimFlags = [
+    "-d:NimblePkgVersion=${finalAttrs.version}"
+  ];
 })
 ```
 
-## Nim library packages in Nixpkgs {#nim-library-packages-in-nixpkgs}
-
+## Lockfiles {#nim-lockfiles}
+Nim lockfiles are created with the `nim_lk` utility.
+Run `nim_lk` with the source directory as an argument and it will print a lockfile to stdout.
+```sh
+$ cd nixpkgs
+$ nix run -f . nim_lk -- $(nix eval --raw -f . ttop.src) > pkgs/by-name/tt/ttop/lock.json
```

Comment:
Would it make sense to add this combined with `nix-update` as a default `passhtru.updateScript` in `buildNimPackage`? Then both r-ryantm and `nix-update -u` would able to bump nim packages without upstream changes to handle nim packages.

Reply:
That should definitly happen, I just haven't looked into it.

Reply:
If it means writing Python then it's not going to me to implement it.

Reply:
Could be something like this, haven't tested
```nix
updateScript = writeShellScript "update-${finalAttrs.finalPackage.pname}" {
  nativeBuildInputs = [ nim_lk nix-update ];
} ''
  nix-update "$UPDATE_NIX_ATTR_PATH"
  src="$(nix build -f . "$UPDATE_NIX_ATTR_PATH".src --no-link --print-out-paths)"
  lockFilePath="''${
    nix eval --impure  --expr 'with import ./. {}; lib.removePrefix "${toString ./.}/" '"$UPDATE_NIX_ATTR_PATH"'.lockFile' | cut -d: -f1
  }"
  nim_lk "$src" > "$lockFilePath"
''
```